pg_auth_mon.c

```diff
@@ -191,6 +194,8 @@ auth_monitor(Port *port, int status)
 		fai->key = key;
 		memset(&fai->total_successful_attempts, 0, sizeof(auth_mon_rec)
 			   - offsetof(auth_mon_rec, total_successful_attempts));
+		fai->user_name = palloc0(sizeof(NameData));
+		strcpy(fai->user_name->data, "dummy_user");
```

Comment:
use namestrcpy. You may also check other_scan in this view, it is also of name type.